meta-lmp-bsp/recipes-bsp/u-boot/u-boot-ostree-scr-fit/apalis-imx6/boot.cmd

```diff
@@ -11,18 +11,153 @@ setenv devnum 0
 setenv fdt_addr 0x15800000
 setenv optee_ovl_addr 0x16000000
 
+setenv bootloader 2
+setenv bootloader2 100
+setenv bootloader_s 802
+setenv bootloader2_s 900
+setenv bootloader_image "SPL"
+setenv bootloader2_image "u-boot.itb"
+
 setenv bootcmd_resetvars 'setenv kernel_image; setenv bootargs; setenv kernel_image2; setenv bootargs2'
 setenv bootcmd_otenv 'run bootcmd_resetvars; ext4load ${devtype} ${devnum}:2 ${loadaddr} /boot/loader/uEnv.txt; env import -t ${loadaddr} ${filesize}'
+setenv bootcmd_bootenv 'setenv bootfirmware_version; ext4load ${devtype} ${devnum}:2 ${loadaddr} ${ostree_root}/usr/lib/firmware/version.txt; env import -t ${loadaddr} ${filesize}'
+setenv bootcmd_getroot 'setexpr ostree_root gsub "^.*ostree=([^ ]*).*$" "\\\\1" "${bootargs}"'
+
+# Env handling
+setenv saveenv_mmc 'if test -z "${board_is_closed}"; then saveenv; fi;'
+
+# U-Boot image update helpers
+setenv uboot_hwpart 1
+setenv load_image 'ext4load ${devtype} ${devnum}:2 ${loadaddr} ${image_path}'
+setenv set_blkcnt 'setexpr blkcnt ${filesize} + 0x1ff && setexpr blkcnt ${blkcnt} / 0x200'
+
+# For eMMC we try to switch hw partition (to write boot images to boot0)
+setenv update_image 'run set_blkcnt && mmc dev ${devnum} ${uboot_hwpart} && mmc write ${loadaddr} ${start_blk} ${blkcnt}'
+
+imx_secondary_boot
+
+run bootcmd_otenv
+run bootcmd_getroot
+
+imx_is_closed
+
+# Board is open, so ubootenv is used
+if test -n "${board_is_closed}"; then
+	echo "Using FIOVB RPMB"
+else
+	echo "Using ubootenv"
+	setenv fiovb.bootcount "${bootcount}"
+	setenv fiovb.rollback "${rollback}"
+	setenv fiovb.upgrade_available "${upgrade_available}"
+	setenv fiovb.bootupgrade_available "${bootupgrade_available}"
+fi
+
+if test "${fiovb.is_secondary_boot}" = "0"; then
+	if test -n "${board_is_closed}"; then
+		# FIO bootcount, upgrade_available and rollback initialization
+		mmc rescan
+		fiovb init {devnum}
```

Comment:
shouldn't this be `fiovb init ${devnum}`

Reply:
yes, will fix